Support entering and saving command line arguments

yoshisuga opened the pull request:
After I made this change, I realized I can adjust those command line options I was interested in in the MAME UI menu. But still, might be useful to have?

ToddLa commented:
Replacing UX that is in the MAMEUI is always a good thing (ie goal). 

For example MAME4iOS should have a native UX for button mapping, even tho you can do it in MAMEUI

ToddLa commented:
Changes look good, I get what you are doing with the "helper class"

* Need to remove the `Info.plist` and `scheme` from the PR, they look accidental. 

* need a way to remove custom cmd line keys from UserDefaults when game is deleted or reset. But this can be done later.

yoshisuga commented:
> Changes look good, I get what you are doing with the "helper class"
> 
> * Need to remove the `Info.plist` and `scheme` from the PR, they look accidental.
> * need a way to remove custom cmd line keys from UserDefaults when game is deleted or reset. But this can be done later.

Reverted those unintended Xcode project/scheme file changes, and added support for deleting from UserDefaults.
